### ISSUE TITLE
Retire `astor`, as `ast.unparse` exists, and migrate it into API-compatible shims of `ast` for older versions of python

### ISSUE THREAD

**Issue by KOLANICH** [berkerpeksag/astor#204]

**Comment by SamuelMarks** [comment]
Yep I've been doing that for ages: https://github.com/offscale/cdd-python/blob/f64f886/cdd/source_transformer.py#L13-L27

**Comment by Kodiologist** [comment]
Hy also does this, since https://github.com/hylang/hy/pull/1999. astor might as well do it internally.

**Comment by leycec** [comment]
**Necrobump.** The `astor.code_to_ast()` function decompiling arbitrary code objects into corresponding ASTs still appears to have *no* equivalent in any other library – including the standard `ast` module. Is that correct? If so, `astor` should absolutely *not* retire but instead mono-focus on this one thing `astor` still does that nobody else does.

In fact, that's exactly how I discovered `astor`: via [this StackOverflow question](https://stackoverflow.com/questions/67922884/is-possibile-to-get-the-ast-from-a-python-code-object), whose first comment namedrops `astor` as the only available solution. Long live `astor`! :fist_raised:

**Comment by akaihola** [comment]
Annoyingly, [`ast.unparse()`] outputs single quoted strings by default while modern code reformatters default to double quotes.

[`ast.unparse()`] behavior is impossible to change since it:
- doesn't provide any options to tweak its behavior
- uses the [`ast._Unparser`] class which is not part of the public API

ikamensh/flynt#195 is currently struggling with this and would benefit from `astor` continuing to be maintained and made compatible with Python 3.14.

[`ast._Unparser`]: https://github.com/python/cpython/blob/5fa7e1b7fd57e8c6297e9eb79d79cede42e5ce0f/Lib/ast.py#L666
[`ast.unparse()`]: https://docs.python.org/3/library/ast.html#ast.unparse

**Comment by Kodiologist** [comment]
> modern code reformatters default to double quotes

That seems like an unforced error. If I recall correctly, Python's `repr` has always used single quotes, and single quotes have always predominated in official documentation. I'm not sure why Black decided to go with double quotes.

**Comment by akaihola** [comment]
Another difference in behavior between `astor` and `ast.unparse()`:
```python
>>> import ast
>>> import astor
>>> node = ast.JoinedStr(
...     values=[ast.FormattedValue(value=ast.IfExp(test=ast.Name(id='c'), body=ast.Name(id='a'), orelse=ast.Name(id='b')), conversion=-1)]
... )
>>> print(ast.unparse(node))
f'{(a if c else b)}'
>>> print(astor.to_source(node))
f"""{a if c else b}"""
```

This is being discussed in ikamensh/flynt#205, preventing migrating from `astor` to `ast.unparse()`.
